hal/src/boron/network/sara_ncp_client.cpp

```diff
@@ -412,6 +412,15 @@ int SaraNcpClient::queryAndParseAtCops(CellularSignalQuality* qual) {
     r = CHECK_PARSER(resp.readResult());
     CHECK_TRUE(r == AtResponse::OK, SYSTEM_ERROR_AT_NOT_OK);
 
+    // Preserve digit format data
+    const int mnc_digits = ::strnlen(mobileNetworkCode, 4);
+    CHECK_TRUE((2 == mnc_digits || 3 == mnc_digits), SYSTEM_ERROR_BAD_DATA);
+    if (2 == mnc_digits) {
+        cgi_.cgi_flags |= CGI_FLAG_TWO_DIGIT_MNC;
+    } else {
+        cgi_.cgi_flags &= ~CGI_FLAG_TWO_DIGIT_MNC;
```

Comment:
Let's zero out `cgi_.cgi_flags`, in which case this becomes unnecessary.

Same here https://github.com/particle-iot/device-os/pull/1804/files#diff-13c08dca4df073df121b6a68266e0105R1524

Reply:
Yeah, we talked about it and I forgot... :+1:

Reply:
Correct me if I'm wrong, but I believe the struct _(and subsequently the `cgi_flags`)_ were already zero'd at [mdm_hal.cpp|Line 1212](https://github.com/particle-iot/device-os/blob/mnc2d/hal/src/electron/modem/mdm_hal.cpp#L1212) for Gen2 devices, and is initialized because it is a non-static data member of `SaraNcpClient` at [sara_ncp_client.h|Line 80](https://github.com/particle-iot/device-os/blob/mnc2d/hal/src/boron/network/sara_ncp_client.h#L80) for Gen3 devices.

I'm less confident for the Gen3 devices, but that was my original understanding...

Reply:
I'm going to explicitly zero the struct on the `SaraNcpClient` to ensure it is behaving as expected.